Refactor doc

benbovy opened the pull request.

willirath commented:
I've modified the rtd webhook to not only trigger on pushes, but also on PRs (wasn't enabled until a few minutes ago). With #25, I get a status report from rtd building the pr. (And as I re-used the commits from this branch, I guess there's a re-computation of all checks here as well?)

willirath commented on doc/api.rst:
```diff
@@ -0,0 +1,60 @@
+.. code-block:: python
+
+   >>> import xarray as xr         # first import xarray
+   >>> import xsimlab              # import xoak (the 'xoak' accessor is registered)
```
```suggestion
   >>> import xoak                 # import xoak (the 'xoak' accessor is registered)
```

benbovy replied:
Ah yes copy/paste remnant.

benbovy commented:
I refactored and moved the notebooks in `doc/examples` so that they can be both run with binder and shown in the documentation.

@willirath if CI passes (including doc build) this is ready to merge for me, if you want to have another look...

willirath commented:
Looks great! Merging.
